### PR TITLE
Feature/ipv4 address removal

### PR DESCRIPTION
**Pull request by metelik**
Adding an ipv4 address removal functionality. 'Twas being invoked from dhcpv4_manager.ex but yet non-existent.

### CODE REVIEW

File: src/netif.c [hunk 1]
```diff
@@ -1762,7 +1866,7 @@ static int add_default_gateway(struct netif *nb, const char *ifname, const char
     int idx = ifname_to_index(nb, ifname);
 
     if (!idx) {
-      debug("[%s %d %s]: No such device: '%s'\r\n", __FILE__, __LINE__, __FUNCTION__, ifname);
+      debug("[%s %d]: No such device: '%s'\r\n", __FILE__, __LINE__, ifname);
```
Comment (adrigonzo) [inline src/netif.c:1875]: why are you removing function name here? Isn't it usually useful for debug logs?

File: src/netif.c [hunk 2]
```diff
@@ -1403,6 +1434,79 @@ static int set_ipaddr_ioctl(const struct ip_setting_handler *handler, struct net
...
+
+  struct sockaddr_in *addr = (struct sockaddr_in *) &ifr.ifr_addr;
+  struct nlmsghdr    *nlh  = mnl_nlmsg_put_header(nb->nlbuf);
+  struct ifaddrmsg   *ipm  = mnl_nlmsg_put_extra_header(nlh, sizeof(struct ifaddrmsg));;
+
+  int seq = nb->seq++;
+  int ret = 0;
+
+  nlh->nlmsg_type  = RTM_DELADDR;
+  nlh->nlmsg_flags = NLM_F_REQUEST;// | NLM_F_ACK;
```
Comment (adrigonzo) [inline src/netif.c]: any reason to keep the commented out flag?

**Review by adrigonzo — APPROVED** [review]

File: src/netif.c (hunk 1 above)
Comment (metelik) [reply src/netif.c:1875]: ISO was bitching about __FUNCTION__ being not part of the standard :(

File: src/netif.c (hunk 2 above)
Comment (metelik) [reply src/netif.c]: fair point :) Re-adding the ACK ;)

**Review by PeterJeaiem — APPROVED** [review]

**Pull request merged**